src/lamplib/src/genny/tasks/auto_tasks_local.py

```diff
@@ -0,0 +1,103 @@
+PRIVATE_WORKLOADS_REPO_URL = "https://github.com/10gen/PrivateWorkloads.git"
+WORKLOAD_FILE_PATTERN = os.path.join("**", "src", "workloads", "**", "*.yml")
+FIRST_EXECUTION = 0
+WORKLOAD_PATH_KEY = "auto_workload_path"
+DSI_TASK_NAME = "f_run_dsi_workload"
+
+PROJECT_FILES = {
+    "master": [
+        os.path.join(DSI_TMP_PATH, "evergreen", "system_perf", "master", "variants.yml"),
+        os.path.join(DSI_TMP_PATH, "evergreen", "system_perf", "master", "master_variants.yml"),
+    ],
+    "8.0": [
+        os.path.join(DSI_TMP_PATH, "evergreen", "system_perf", "8.0", "variants.yml"),
+    ],
+    "7.3": [
+        os.path.join(DSI_TMP_PATH, "evergreen", "system_perf", "7.3", "variants.yml"),
+    ],
+    "7.0": [
+        os.path.join(DSI_TMP_PATH, "evergreen", "system_perf", "7.0", "variants.yml"),
+    ],
+    "6.0": [
+        os.path.join(DSI_TMP_PATH, "evergreen", "system_perf", "6.0", "variants.yml"),
+    ],
+    "5.0": [
+        os.path.join(DSI_TMP_PATH, "evergreen", "system_perf", "5.0", "variants.yml"),
+    ],
+}
+
+
+def get_builds(branch_name: str):
+    builds = []
+    expansions = {"branch_name": branch_name, "execution": FIRST_EXECUTION}
+    for project_file in PROJECT_FILES[branch_name]:
+        builds.extend(get_all_builds(expansions, project_file))
+
+    return builds
+
+
+def fix_auto_workload_path(command: CommandDefinition):
+    if command._vars[WORKLOAD_PATH_KEY].startswith("src"):
+        command._vars[WORKLOAD_PATH_KEY] = "src/genny/" + command._vars[WORKLOAD_PATH_KEY]
+    if command._vars[WORKLOAD_PATH_KEY].startswith("PrivateWorkloads"):
+        command._vars[WORKLOAD_PATH_KEY] = "src/" + command._vars[WORKLOAD_PATH_KEY]
+
+
+def main(workspace_root: str, running_in_evergreen: bool) -> None:
+    if not running_in_evergreen:
+        SLOG.info("Cloning the DSI repo to look for variants")
+        GitRepo.clone_from(DSI_REPO_URL, DSI_TMP_PATH)
+        SLOG.info("Cloning the PrivateWorkloads repo to look for tests")
+        GitRepo.clone_from(PRIVATE_WORKLOADS_REPO_URL, PRIVATE_WORKLOADS_TMP_PATH)
+
+    for branch_name in PROJECT_FILES.keys():
+        SLOG.info("Creating the configuration", branch_name=branch_name)
+        builds = get_builds(branch_name)
+        lister = WorkloadLister(
+            workspace_root=workspace_root, workload_file_pattern=WORKLOAD_FILE_PATTERN
+        )
+        repo = Repo(lister=lister, reader=YamlReader(), workspace_root=workspace_root)
+        config = create_configuration(repo, builds, False, [])
+
+        SLOG.info("Removing global exec_timeout_secs as it is already in the main project")
+        config._exec_timeout_secs = None
+
+        SLOG.info("Fixing the auto_workload_path to match the DSI folder structure")
+        for task in config._tasks:
+            for command in task._commands._cmd_seq:
+                if command._function_name == DSI_TASK_NAME:
+                    fix_auto_workload_path(command)
```

Comment:
I feel like there's a a number of these hacks scattered around because all the tooling is expecting a directory structure of the form:
```
src
  genny/src/workloads/**
  PrivateWorkloads/src/workloads/**
```

but you're checking out things as private child directories of the genny repo.

Since you're making temporary folders anyway, you could probably get this all working using a couple of simlinks and avoid some headache when running in evergreen.

Specifically, you could make a temporary directory and symlink any relevant folders to get a similar directory structure, and cut down on the fixing up.

Reply:
I've hit a few bumps with this/tried a few things:
* Cloning the private workloads repo next to the genny repo does not work, as this would modify something outside the repo, which is a bad practice.
* Symlinking the folders to `src/*/workloads` inside the genny repo is not very clean, as there is already a src/genny folder in the genny repo.
* Symlinking the folders to a temporary folder does not work as some of the code call git commands, meaning we need to be inside the correct git folder. 
* Symlinking the folders to some other folder in the genny folder results in a prefix we need to remove again.

I don't think anything here is insurmountable, but the effort/payoff ratio for something that will hopefully only live for ~6 months does not seem worth it. Let me know if you disagree with this :)